### DISCUSSION TITLE
Add the ability to change the initial query in `consult-completion-in-region`

### DISCUSSION THREAD

**Discussion by NightMachinery** [minad/consult#956] [Q&A] [answered]
I like to add a space to my completion queries, as I use `orderless`. Looking at the code, I think the line `:initial initial` should be changed to:
```
:initial (consult--completion-transform-initial initial)
```
with `consult--completion-transform-initial` being set to the identity function by default.

**Comment by oantolin** [comment]
Huh, this is an interesting idea. I also use `orderless` and `consult-completion-in-region` and I don't always start by typing `SPC`, but it's true that I frequently do. Maybe I'd like it better if a space were always added and I sometimes deleted it. I think I'd need to try that for a while to know if it suited me better, but it's definitely a reasonable idea.

**Comment by minad** [comment, marked as answer]
Indeed, this could makes sense. Right now we support overriding the `:initial` string via `consult-customize` but not transforming the string. Maybe we find a general way to neatly add this to the existing commands. However I wonder if this feature would end up being used much. As @oantolin mentioned he does not always add SPC, just frequently.

Maybe you also want to emulate `ivy-initial-inputs-alist` for other commands? So far I've considered this feature out of scope of Vertico and Ivy. I don't plan to add every customization option from Ivy to Vertico, but a lot of features can be implemented via the general `vertico-multiform-mode`, also this one:

~~~ elisp
(setf (alist-get #'consult-completion-in-region vertico-multiform-commands)
      (list (lambda (_) (insert " "))))
~~~

So maybe the solution via `vertico-multiform-mode` is already good enough?

**Reply by NightMachinery** [reply]
Thanks. This `vertico-multiform-mode` looks great and it works. I think it runs an arbitrary command after vertico opens up? If so, we don't really need things like `ivy-initial-inputs-alist`.
